core/test/components/range_minimum_query.cpp

```diff
@@ -0,0 +1,73 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "core/components/range_minimum_query.hpp"
+
+#include <gtest/gtest.h>
+
+#include "core/test/utils.hpp"
+
+
+namespace {
+
+
+TEST(RangeMinimumQuery, RepresentativesAreExhaustive)
+{
+    constexpr auto size = 8;
+    using tree = gko::detail::cartesian_tree<size>;
+    int values[size]{};
```

Comment:
That's a technicality about constructing vs. assigning to uninitialized elements that I wanted to avoid. This kind of initialization is required in constexpr code, which is why I used it here as well.